File: tutorial/abs/ABC086C.php
<?php
    fscanf(STDIN,"%d",$n);
    $t_p = 0;
    $x_p = 0;
    $y_p = 0;
    $result = "Yes";
    for($i=0;$i<$n;$i++){
        fscanf(STDIN,"%d %d %d",$t,$x,$y);
        if(abs($t-$t_p) < abs($x-$x_p)+abs($y-$y_p) || abs($t-$t_p)%2 != (abs($x-$x_p)+abs($y-$y_p))%2) {
            $result = "No";
            break;
        }
        $t_p = $t;
        $x_p = $x;
        $y_p = $y;
    }
    echo $result;

File: tutorial/abs/ABC086C.py
n = int(input())

x_p,y_p,t_p = 0,0,0
result = "Yes"
for i in range(n):
   t,x,y = map(int,input().split())
   if(abs(t-t_p) < abs(x-x_p)+abs(y-y_p) or abs(t-t_p)%2 != (abs(x-x_p)+abs(y-y_p))%2):
       result = "No"
       break
   t_p,x_p,y_p = t,x,y

print(result)
